Spigot-Server-Patches/0747-Add-feature-access-to-custom-ChunkGenerators.patch

```diff
@@ -0,0 +1,167 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: dfsek <dfsek@protonmail.com>
+Date: Tue, 25 May 2021 17:09:34 -0700
+Subject: [PATCH] Add feature access to custom ChunkGenerators
+
+
+diff --git a/src/main/java/io/papermc/paper/world/CraftProtoWorld.java b/src/main/java/io/papermc/paper/world/CraftProtoWorld.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..e037126f1fd2af9bd196c57e9af5856dc6657318
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/world/CraftProtoWorld.java
+@@ -0,0 +1,97 @@
++package io.papermc.paper.world;
++
++import com.google.common.base.Preconditions;
++import net.minecraft.core.BlockPosition;
++import net.minecraft.server.level.RegionLimitedWorldAccess;
++import net.minecraft.world.entity.EntityInsentient;
++import net.minecraft.world.entity.EnumMobSpawn;
++import net.minecraft.world.level.block.BlockFluids;
++import org.bukkit.World;
++import org.bukkit.block.data.BlockData;
++import org.bukkit.craftbukkit.block.data.CraftBlockData;
++import org.bukkit.entity.Entity;
++import org.bukkit.event.entity.CreatureSpawnEvent;
++import org.bukkit.util.Consumer;
++import org.bukkit.util.Vector;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++import java.lang.ref.WeakReference;
++import java.util.Objects;
++
++public class CraftProtoWorld implements ProtoWorld {
++    /*
++     * Keep the world in a weak reference to prevent people from doing dumb
++     * things that keep RLWA's around after we're done with them.
++     */
++    private final WeakReference<RegionLimitedWorldAccess> world;
```

Comment:
there's no point in this being a weakref if you're clearing it immediately after use